### ISSUE TITLE
CI unrelated failures

### ISSUE THREAD

**Issue by wxiaoguang** [go-gitea/gitea#18077]
During unit-test:

```
+ make unit-test-coverage test-check
Running unit-test-coverage -race -tags 'sqlite sqlite_unlock_notify'...
?   	code.gitea.io/gitea	[no test files]
ok  	code.gitea.io/gitea/build/codeformat	0.189s	coverage: 71.8% of statements
ok  	code.gitea.io/gitea/cmd	0.793s	coverage: 8.6% of statements
?   	code.gitea.io/gitea/contrib/environment-to-ini	[no test files]
......
?   	code.gitea.io/gitea/services/task	[no test files]
ok  	code.gitea.io/gitea/services/user	6.467s	coverage: 30.7% of statements
ok  	code.gitea.io/gitea/services/webhook	4.276s	coverage: 62.0% of statements
ok  	code.gitea.io/gitea/services/wiki	13.117s	coverage: 65.5% of statements

==>Ok

Running test-check...
make test-backend has changed files in the source tree:
?? mssql-log/
?? mysql-log/
?? mysql8-log/
You should change the tests to create these files in a temporary directory.
Do not simply add these files to .gitignore
make: *** [Makefile:368: test-check] Error 1
```

**Comment by zeripath** [comment]
I think this indicates that you've ran some other tests (Likely the integration tests `make test-mssql` etc.) which have created those log files.

The purpose of the make test-check is to detect if test-backend has created any files only and it assumes that there are not any uncommited files at the start of the test. It's not trying to be clever.

Therefore we have four solutions:

* Change the wording here to make it less certain and say there are uncommitted files - if these were present before you ran test-backend there's nothing to worry about but otherwise you need to find out what has happened.
* Add these to .gitignore
* Change the integrations tests to make sure that they put their logs with the rest of their data, or put them in a new directory e.g. integrations-test-logs which can be ignored.
* Make the test a bit cleverer so it only detects newly changed files.

I think however as this is only a rough and ready test to detect if test-backend is doing naughty things on drone I'm not sure any of this is worth doing.

**Comment by wxiaoguang** [comment]
Hmm ... it happens during our CI, and only happens randomly, I haven't caught the reason or investigated. 

Among these solutions, which one is best?

**Comment by zeripath** [comment]
oh that's interesting.

I guess we should change the drone configuration to make sure that these logs end up on the temp dir.

I'll take a look at this.

**Comment by zeripath** [comment]
I've folded the fix in to #18447
